### ISSUE TITLE
Potential raciness on unbalanced-tuple-unpacking

### ISSUE THREAD

**Issue by schopin-pro** [pylint-dev/pylint#8844]
### Bug description

In the [apport project](https://github.com/canonical/apport), among others we have a small Python script that has code looking like this:
```python
if len(sys.argv) != 3:
    sys.exit(1)

execname, sourcefile) = sys.argv[1:3]
```

We recently enable parallelism for our Pylint runs, and since then we're sometime triggering `unbalanced-tuple-unpacking` on the last line:
data/gcc_ice_hook:28:0: W0632: Possible unbalanced tuple unpacking with sequence: left side has 2 label(s), right side has 0 value(s) (unbalanced-tuple-unpacking)
(see https://github.com/canonical/apport/actions/runs/5532030665/jobs/10093530885 if you have access to the logs?)

### Command used

```shell
pylint -j 0 ${PYTHON_FILES} ${PYTHON_SCRIPTS}
```

### Pylint output

```shell
************* Module gcc_ice_hook
data/gcc_ice_hook:28:0: W0632: Possible unbalanced tuple unpacking with sequence: left side has 2 label(s), right side has 0 value(s) (unbalanced-tuple-unpacking)
```

### Expected behavior

I wouldn't expect any warning, the code is obviously correct. Or at least I'd expect the warning to show up consistently.

### Pylint version

```shell
pylint installed from the system packages:

pylint 2.12.2
astroid 2.9.3
Python 3.10.6 (main, May 29 2023, 11:10:38) [GCC 11.3.0]
```

### OS / Environment

Ubuntu Jammy 22.04

Also happens on Kinetic, so far hasn't happend on Lunar (pylint 2.15)

**Comment by Pierre-Sassoulas** [comment]
Thank you for opening the issue. This is a duplicate (but I'm on mobile so linking is hard). 'sys.argv' can't be inferred because the value given during static analysis to pylint will obviousely not be the same at runtime. There's a fix under review in astroid for this.

**Comment by schopin-pro** [comment]
This duplicates #7710 indeed. For some reason it didn't turn up in my initial search (probably a typo in my search params?). Thanks for the diligent response!
